sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/siprouting/implementation/SipRoutingAdminClientImpl.java

```diff
@@ -0,0 +1,390 @@
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<SipConfiguration>> getSipConfigurationWithResponseAsync() {
+        final String accept = "application/json";
+        return FluxUtil.withContext(
+                context -> service.getSipConfiguration(this.getEndpoint(), this.getApiVersion(), accept, context));
+    }
+
+    /**
+     * Gets SIP configuration for resource.
+     *
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return sIP configuration for resource.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<SipConfiguration>> getSipConfigurationWithResponseAsync(Context context) {
+        final String accept = "application/json";
+        return service.getSipConfiguration(this.getEndpoint(), this.getApiVersion(), accept, context);
+    }
+
+    /**
+     * Gets SIP configuration for resource.
+     *
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return sIP configuration for resource.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SipConfiguration> getSipConfigurationAsync() {
+        return getSipConfigurationWithResponseAsync()
+                .flatMap(
+                        (Response<SipConfiguration> res) -> {
+                            if (res.getValue() != null) {
+                                return Mono.just(res.getValue());
+                            } else {
+                                return Mono.empty();
+                            }
+                        });
+    }
+
+    /**
+     * Gets SIP configuration for resource.
+     *
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws CommunicationErrorResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return sIP configuration for resource.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<SipConfiguration> getSipConfigurationAsync(Context context) {
+        return getSipConfigurationWithResponseAsync(context)
+                .flatMap(
+                        (Response<SipConfiguration> res) -> {
+                            if (res.getValue() != null) {
+                                return Mono.just(res.getValue());
+                            } else {
+                                return Mono.empty();
+                            }
```

Comment:
You may want to use a newer version of Autorest, I see the Swagger README shows `4.0.40` being used, which offers cleaner code generation, this whole block is replaceable and replaced with `Mono.justOrEmpty(res.getValue())`, fixes, and performance optimizations. The last version in `4.0.x` is `4.0.62` and the newest release is `4.1.2`.

Reply:
Upgraded to @autorest/java@4.1.3 https://github.com/Azure/azure-sdk-for-java/pull/26819/commits/723d49e7fca90431bffa3366f19bcad1d20ddfba#diff-c765ec1e0369684977f9ebf8c26b5855dc18c9446b446cd1c916f2cc267deb3d